sensors/trajectory_calc.py

```diff
@@ -147,13 +162,26 @@ def get_rotation_to_speaker(self):
         robot_pose_2d = self.odometry.getPose()
         robot_to_speaker = speaker_translation - robot_pose_2d.translation()
         return robot_to_speaker.angle()
+    
+    def get_rotation_to_speaker_moving(self):
+        speaker_translation:Translation2d = POI.Coordinates.Structures.Scoring.kSpeaker.getTranslation()
+        t_total = self.get_distance_to_target() / (self.v0_effective * np.cos(self.get_theta()))  if self.v0_effective > 0 else 0
+        
+        
+        rvels = self.get_drivetrain_speeds_field_origin()
+        
+        robot_pose_2d = self.odometry.getPose()
+        robot_pose_2d_w_speeds = robot_pose_2d + Transform2d(rvels.vx * t_total, rvels.vy * t_total, 0)
+        robot_to_speaker = speaker_translation - robot_pose_2d_w_speeds.translation()
+        return robot_to_speaker.angle()
```

Comment:
Not sure what this is all about. Can you explain it to me? Again. I think that we need to be thinking about where we WILL need to be aiming, rather than should be aiming right now. So we may need to be thinking ahead with Kalman filter thing.